mesonpy/__init__.py

```diff
@@ -1009,19 +1010,21 @@ def _get_ninja_if_needed() -> List[str]:
         try:
             candidate_version = tuple(int(x) for x in result.stdout.split('.')[:3])
         except ValueError:
-            # The meson search function skips forward if the version is not readable or too low
             continue
-        if candidate_version < (1, 8, 2):
+        if candidate_version < _NINJA_REQUIRED_VERSION:
             continue
-        return []
+        return True
 
-    return [_depstr.ninja]
+    return False
 
 
 def get_requires_for_build_sdist(
     config_settings: Optional[Dict[str, str]] = None,
 ) -> List[str]:
-    return _get_ninja_if_needed()
+    dependencies = []
+
+    if not _env_has_ninja_command():
+        dependencies.append(_depstr.ninja)
```

Comment:
This looks wrong:
```python
    if not _env_has_ninja_command():
        return _depstr.ninja
```

Reply:
You can't return a string here, it has to be a list. I did rewrite it though to look nicer.

Reply:
Which string?

Reply:
You fixed it. :)

Reply:
😄